src/zap.ts

```diff
@@ -364,6 +388,555 @@ export class Zap {
 
   /////// ZAPIN FUNCTION ////////
 
+  async getZapInDammV2DirectPoolParams(
```

Comment:
this function is only work to zap from SOL, can we make it general, Ex: when user want to zap in from USDC, or from any token?